pkg/client/unversioned/conditions.go

```diff
@@ -168,3 +171,125 @@ func DeploymentHasDesiredReplicas(c ExtensionsInterface, deployment *extensions.
 			deployment.Status.UpdatedReplicas == deployment.Spec.Replicas, nil
 	}
 }
+
+// ErrPodCompleted is returned by PodRunning or PodContainerRunning to indicate that
+// the pod has already reached completed state.
+var ErrPodCompleted = fmt.Errorf("pod ran to completion")
```

Comment:
It's very frustrating because basically these functions work no matter which typed client you're importing :/